### PR TITLE
Enable wight-free export for VLM

### PR DESCRIPTION
**Pull request by quic-amitraj**
## Summary

This PR enables the VLM weight-free export path for `Qwen/Qwen3-VL-30B-A3B-Instruct`, a Qwen3-VL-MoE image-text-to-text model.

Model validated:

```text
Qwen/Qwen3-VL-30B-A3B-Instruct
```

Validation script:

```text
examples/dynamo/image_text_to_text/qwen3_vl_moe_dynamo_inference.py
```
## Performance Comparison

Measured with the full-layer model using:

```bash
python examples/dynamo/image_text_to_text/qwen3_vl_moe_dynamo_inference.py
```

Legacy means no weight-free export. Weight-free uses the Dynamo/`torch.export` path because weight-free export is implemented on top of Dynamo export.

### Export Time

| Component | Legacy | Weight-Free Dynamo | Saved | Improvement |
| --- | ---: | ---: | ---: | ---: |
| Vision export | 28.685 sec | 17.584 sec | 11.101 sec | 1.63x faster, 38.70% lower |
| Decoder export | 1352.880 sec | 86.967 sec | 1265.913 sec | 15.56x faster, 93.57% lower |
| Total export | 1381.565 sec | 104.551 sec | 1277.014 sec | 13.21x faster, 92.43% lower |

### Compile Time

| Component | Legacy | Weight-Free Dynamo | Saved | Improvement |
| --- | ---: | ---: | ---: | ---: |
| Vision compile | 19.051 sec | 18.644 sec | 0.407 sec | 1.02x faster, 2.14% lower |
| Decoder compile | 1705.522 sec | 366.976 sec | 1338.546 sec | 4.65x faster, 78.48% lower |
| Total compile | 1724.573 sec | 385.620 sec | 1338.953 sec | 4.47x faster, 77.64% lower |

## Artifact Size Comparison

For ONNX size, this table uses the complete wrapper export directory excluding QPC, because legacy export stores large external tensor files beside the small `.onnx` protobuf.

### ONNX Export Artifact Size

| Component | Legacy | Weight-Free Dynamo | Reduction |
| --- | ---: | ---: | ---: |
| Vision ONNX artifact | 2.01 GiB | 1.48 MiB | 1386.14x smaller, 99.928% lower |
| Decoder ONNX artifact | 113.81 GiB | 76.92 MiB | 1515.12x smaller, 99.934% lower |
| Total ONNX artifact | 115.82 GiB | 78.40 MiB | 1512.68x smaller, 99.934% lower |

### QPC Size

| Component | Legacy | Weight-Free Dynamo | Delta |
| --- | ---: | ---: | ---: |
| Vision QPC | 1.04 GiB | 1.04 GiB | 97.65 KiB smaller |
| Decoder QPC | 44.55 GiB | 44.55 GiB | 2.25 MiB larger |
| Total QPC | 45.59 GiB | 45.59 GiB | 2.15 MiB larger |

The QPC size is effectively unchanged, as expected. The major storage improvement is in the exported ONNX artifact before compile.

## Runtime Comparison

| Metric | Legacy | Weight-Free Dynamo | Delta |
| --- | ---: | ---: | ---: |
| Average prefill / TTFT | 1.03 sec | 1.03 sec | no change |
| Decode throughput | 66.49 tok/sec | 65.60 tok/sec | -1.34% |
| Total throughput | 39.59 tok/sec | 39.30 tok/sec | -0.73% |
| Total E2E inference time | 2.50 sec | 2.52 sec | +0.80% |

Runtime is effectively at parity while export, compile, and pre-compile artifact size improve substantially.

### CODE REVIEW

**Comment by quic-amitraj** [comment]
Legacy output:

['Based on the image provided, the following colors are visible:\n\n-   **Black:** This is the dominant color, seen in the fur of the puppy.\n-   **Brown:** The wooden surface has various shades of brown, from light tan to darker, weathered brown.\n-   **Gray:** There are significant gray tones in the wood, particularly in the shadows and the weathered parts of the planks.\n-   **White:** The highlights in the puppy\'s eyes appear as bright white or"']

Weight-free output:

['Based on the image provided, the following colors are visible:\n\n-   **Black:** This is the dominant color, seen in the fur of the puppy.\n-   **Brown:** The wooden surface has various shades of brown, from light tan to darker, weathered brown, creating a rustic appearance.\n-   **Gray:** There are noticeable gray tones in the wood, particularly in the shadows and the weathered parts of the planks.\n-   **White:** The highlights in the puppy\'s eyes"']
